core/perf_test/CMakeLists.txt

```diff
@@ -128,7 +128,7 @@ kokkos_add_benchmark(PerformanceTest_Benchmark SOURCES ${BENCHMARK_SOURCES})
 kokkos_add_benchmark(Benchmark_Atomic_MinMax SOURCES test_atomic_minmax_simple.cpp)
 
 # FIXME_NVHPC
-if(NOT KOKKOS_CXX_COMPILER_ID STREQUAL NVHPC)
+if(NOT KOKKOS_CXX_COMPILER_ID STREQUAL NVHPC AND Kokkos_ENABLE_DEPRECATED_CODE_4)
   kokkos_add_benchmark(PerformanceTest_Mempool SOURCES test_mempool.cpp)
 endif()
```

Comment:
You don't have to disable deprecation warnings in `test_mempool.cpp`?

Reply:
Yes I do 
```
/var/jenkins/workspace/Kokkos_PR-7431/core/perf_test/test_mempool.cpp:33:28: error: 'template<class Device> class Kokkos::MemoryPool' is deprecated [-Werror=deprecated-declarations]
 using MemoryPool = Kokkos::MemoryPool<ExecSpace>;
                            ^~~~~~~~~~
```
I find it a bit scary that only the GCC 8.4 build on Jenkins failed